app/controllers/serviceLearning/routes.py

```diff
@@ -27,38 +28,72 @@ def serviceCourseManagement(username=None):
         term = Term.get(Term.id==request.form.get("term"))
         status = CourseStatus.get(CourseStatus.status == "Pending")
-        course = Course.create(
-            courseName=request.form.get("courseName"),
-            courseAbbreviation=request.form.get("courseAbbreviation"),
-            courseCredit=request.form.get("credit"),
-            isRegularlyOccuring=1 if request.form.get("regularOccurenceToggle") else 0,
-            term=term,
-            status=status,
-            createdBy=g.current_user,
-            isAllSectionsServiceLearning=1 if request.form.get("slSectionsToggle") else 0,
-            serviceLearningDesignatedSections=request.form.get("slDesignation"),
-            isPermanentlyDesignated=1 if request.form.get("permanentDesignation") else 0,
-        )
-        for i in range(1, 7):
-            CourseQuestion.create(
-                course=course,
-                questionContent=request.form.get(f"{i}"),
-                questionNumber=i
+        coursecheck = Course.get_or_none(Course.id == request.form.get("courseID"))
+        if coursecheck == None:
+            course = Course.create(
+                courseName=request.form.get("courseName"),
+                courseAbbreviation=request.form.get("courseAbbreviation"),
+                courseCredit=request.form.get("credit"),
+                isRegularlyOccuring=1 if request.form.get("regularOccurenceToggle") else 0,
+                term=term,
+                status=status,
+                createdBy=g.current_user,
+                isAllSectionsServiceLearning=1 if request.form.get("slSectionsToggle") else 0,
+                serviceLearningDesignatedSections=request.form.get("slDesignation"),
+                isPermanentlyDesignated=1 if request.form.get("permanentDesignation") else 0,
             )
-        for instructor in instructorsDict["instructors"]:
-            CourseInstructor.create(course=course, user=instructor.username)
-        flash("New service-learning course successfully created.", "success")
-        return redirect('/serviceLearning/courseManagement')
-
+            for i in range(1, 7):
+                CourseQuestion.create(
+                    course=course,
+                    questionContent=request.form.get(f"{i}"),
+                    questionNumber=i
+                )
+            for instructor in instructorsDict["instructors"]:
+                CourseInstructor.create(course=course, user=instructor.username)
+            return redirect('/serviceLearning/courseManagement')
+        else:
+            updateCourse(request.form.copy(), instructorsDict)
+            return redirect('/serviceLearning/courseManagement')
     terms = Term.select().where(Term.year >= g.current_term.year)
-    return render_template('serviceLearning/slcNewProposal.html', terms=terms)
+    courseData = None
+    return render_template('serviceLearning/slcNewProposal.html', terms=terms, courseData = courseData)
```

Comment:
I would get this logic (create or update) into a function outside the controller so it's easier to test.

Reply:
I meant all of it, including course creation.